reference/tools/data-migration/features/online-ddl-scheme.md

```diff
@@ -0,0 +1,188 @@
+
+## 配置
+
+**online-ddl-scheme** 在 task 配置文件里面与 name 同级，例子详见下面配置 Example (完整的配置及意义可以参考 [DM 完整配置文件示例](/reference/tools/data-migration/configure/task-configuration-file-full.md#完整配置文件示例)：
+
+```yml
+# ----------- 全局配置 -----------
+## ********* 基本信息配置 *********
+name: test                      # 任务名称，需要全局唯一
+task-mode: all                  # 任务模式，可设为 "full"、"incremental"、"all"
+is-sharding: true               # 是否为分库分表合并任务
+meta-schema: "dm_meta"          # 下游储存 `meta` 信息的数据库
+remove-meta: false              # 是否在任务同步开始前移除该任务名对应的 `meta`（`checkpoint` 和 `onlineddl` 等）。
+enable-heartbeat: false         # 是否开启 `heartbeat` 功能
+online-ddl-scheme: "gh-ost"     # 目前仅支持 gh-ost 、pt
+
+target-database:                # 下游数据库实例配置
+  host: "192.168.0.1"
+  port: 4000
+  user: "root"
+  password: ""                  # 如果不为空则需经过 dmctl 加密
+```
+
+## online-schema-change : gh-ost
+
+gh-ost 在实现 online schema change 的过程会产生 3 种 table：
+
+- **gho** 用于应用 DDL ，待数据同步追上 origin table 之后会通过 rename 的方式替换 origin table 。
+- **ghc** 用于存放 online schema change 相关的信息。
+- **del** 表是由 origin table rename 过来的。
+
+dm 在同步过程中会把上述 table 中会分成 3 类：
+
+- ghostTable : \_\*\_gho
+- trashTable : \_\*\_ghc 、\_\*\_del
+- realTable : 执行的 online-ddl 的 origin table
+
+**gh-ost** 涉及的主要 SQL如下：
+
+```sql
+-- 1.
+   Create /* gh-ost */ table `test`.`_test4_ghc` (
+                        id bigint auto_increment,
+                        last_update timestamp not null DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
+                        hint varchar(64) charset ascii not null,
+                        value varchar(4096) charset ascii not null,
+                        primary key(id),
+                        unique key hint_uidx(hint)
+                ) auto_increment=256 ;
+
+-- 2.
+   Create /* gh-ost */ table `test`.`_test4_gho` like `test`.`test4` ;
+
+-- 3.
+   Alter /* gh-ost */ table `test`.`_test4_gho` add column cl1 varchar(20) not null ;
+
+-- 4.
+   Insert /* gh-ost */ into `test`.`_test4_ghc`;
+
+-- 5.
+   Insert /* gh-ost `test`.`test4` */ ignore into `test`.`_test4_gho` (`id`, `date`, `account_id`, `conversion_price`, `ocpc_matched_conversions`, `ad_cost`, `cl2`)
+      (select `id`, `date`, `account_id`, `conversion_price`, `ocpc_matched_conversions`, `ad_cost`, `cl2` from `test`.`test4` force index (`PRIMARY`)
+        where (((`id` > _binary'1') or ((`id` = _binary'1'))) and ((`id` < _binary'2') or ((`id` = _binary'2')))) lock in share mode
+      )   ;
+
+-- 6.
+   Rename /* gh-ost */ table `test`.`test4` to `test`.`_test4_del`, `test`.`_test4_gho` to `test`.`test4`;
+```
+
+> **注意：**
+>
+> 具体 gh-ost 的 SQL 会根据工具执行时所带的参数而变化。本文只列出主要的 SQL ，具体可以参考 gh-ost 官方文档。
+
+### DM 对于 **online-ddl-scheme: gh-ost** 的处理
+
+1. 不执行 _test4_ghc 的创建操作。
+2. 不执行 _test4_gho 的创建操作,根据 ghost_schema 、 ghost_table 以及 dm_worker 的 server_id 删除下游 dm_meta.{task_name}\_onlineddl 的记录，清理内存中的相关信息。
+
+   ```sql
```

Comment:
If we add this `copyable` snippet before code blocks, readers will be able to copy the code directly from the website.
```suggestion

{{< copyable "sql" >}}

   ```sql
```

Reply:
These are internal sqls of online schema tools .  I think the readers don't want to copy.